plugins/bufferlines/barbar.nix

```diff
@@ -69,249 +24,567 @@ let
+            };
+            deleted = {
+              enabled = true;
+              icon = "-";
+            };
+          }
+          "Gitsigns icons.";
+
+      filename = helpers.defaultNullOpts.mkBool true ''
+        If `true`, show the name of the file.
+      '';
+
+      filetype = {
+        custom_colors = helpers.defaultNullOpts.mkBool false ''
+          If `true`, the `Buffer<status>Icon` color will be used for icon colors.
+        '';
+
+        enabled = helpers.defaultNullOpts.mkBool true ''
+          Filetype `true`, show the `devicons` for the associated buffer's `filetype`.
+        '';
+      };
+
+      separator = {
+        left = helpers.defaultNullOpts.mkStr "▎" ''
+          The left separator between buffers in the tabline.
+        '';
+
+        right = helpers.defaultNullOpts.mkStr "" ''
+          The right separator between buffers in the tabline.
+        '';
+
+        separator_at_end = helpers.defaultNullOpts.mkBool true ''
+          If true, add an additional separator at the end of the buffer list.
+          Can be used to create a visual separation when the inactive buffer background color is the
+          same as the fill region background color.
+        '';
+      };
+
+      modified = helpers.defaultNullOpts.mkAttrsOf types.anything { button = "●"; } ''
+        The icons which should be used for a 'modified' buffer.
+        Supports all the base options (e.g. `buffer_index`, `filetype.enabled`, etc).
+      '';
+
+      pinned =
+        helpers.defaultNullOpts.mkAttrsOf types.anything
+          {
+            button = false;
+            filename = false;
+            separator.right = " ";
+          }
+          ''
+            The icons which should be used for a pinned buffer.
+            Supports all the base options (e.g. `buffer_index`, `filetype.enabled`, etc).
+          '';
+
+      alternate = helpers.mkNullOrOption (with types; attrsOf anything) ''
+        The icons which should be used for the `|alternate-file|`.
+        Supports all the base options (e.g. `buffer_index`, `filetype.enabled`, etc) as well as
+        `modified` and `pinned`.
+      '';
+
+      current = helpers.mkNullOrOption (with types; attrsOf anything) ''
+        The icons which should be used for current buffer.
+        Supports all the base options (e.g. `buffer_index`, `filetype.enabled`, etc) as well as
+        `modified` and `pinned`.
+      '';
+
+      inactive =
+        helpers.defaultNullOpts.mkAttrsOf types.anything
+          {
+            separator = {
+              left = "▎";
+              right = "";
+            };
+          }
+          ''
+            The icons which should be used for `|hidden-buffer|`s and `|inactive-buffer|`s.
+            Supports all the base options (e.g. `buffer_index`, `filetype.enabled`, etc) as well as
+            `modified` and `pinned`.
+          '';
+
+      visible = helpers.mkNullOrOption (with types; attrsOf anything) ''
```

Comment:
Why not use `mkAttrsOf`?

Reply:
Because I was not able to guess the plugin default value. So I preferred to have nothing.

Reply:
Makes sense. In that case, maybe add a comment to each place where this applies?

Reply:
This needs a `maybeRaw` no?

Reply:
`maybeRaw (attrsOf anything)`

Reply:
Done. The documentation already mentions "Supports all the base options (e.g. `buffer_index`, `filetype.enabled`, etc) as well as `modified` and `pinned`.".